packages/express/lib/run.js

```diff
@@ -30,16 +47,25 @@ const getPort = (port) => {
 };
 
 module.exports = (app, { config, inspectServer, handleError }) => {
+  const { domain } = app.locals;
   const { host, port, https } = config;
   const server = createServer(app, https);
-  getPort(port).then((port) =>
-    server.listen(port, host, (error) => {
-      if (error) {
-        server.close();
-        handleError(error);
-      } else {
-        inspectServer(server);
-      }
-    })
+  domain.on('error', (error) =>
+    shutdown(app).then(() => handleError(error), () => handleError(error))
```

Comment:
If shutting down the app on `SIGTERM` errors, will this trigger the `'error'`-callback of the `domain` and thus invoke a second shutdown of the app?

Reply:
Forget what I wrote. I got `handleError()` wrong.